### ISSUE TITLE
False positive in `no-unused-definitons` rule when using mkdocs admonitions

### ISSUE THREAD

**Issue by samjcombs** [remarkjs/remark-lint#307]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

`no-unused-definitions`

### Steps to reproduce

When using [mkdoc admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/), given this input: 
```markdown
1    !!! note
2    
3       Here is a note with a [reference].
4    
5    [reference]: http://is-properly-defined/
```

an warning is reported on L5 

### Expected behavior

No warning should be reported

### Actual behavior

A warning is reported, possibly due to ast being wierd inside the note admonition block?

**Comment by ChristianMurphy** [comment]
Welcome @samjcombs!
Sorry you ran into a spot of trouble.
Could you share a minimal reproducible example of what you are seeing? (https://stackoverflow.com/help/minimal-reproducible-example)

I tried to re-create the issue in CodeSandbox using `remark` and `remark-lint-no-unused-definitions` programmatically, and displaying any error messages, and don't see an error being reported.
runnable example: https://codesandbox.io/s/lively-morning-qcis7e?file=/src/index.js

<details>
<summary>source from sandbox</summary>

```js
// ⚠️ Important! Please make sure the dependencies are up to date.
// On code sandbox, you can refresh them in the Dependencies section (left-bottom)
// On stackblitz, you can open the package.json file, update the versions,
// then run npm install in the stackblitz terminal

import { remark } from "remark";
import noUnusedDefinitions from "remark-lint-no-unused-definitions";

const sourceMarkdown = `
!!! note

   Here is a note with a [reference].

[reference]: http://is-properly-defined/
`;

async function main() {
  document.querySelector("#source").textContent = sourceMarkdown;

  const file = await remark()
    .use(noUnusedDefinitions)
    // .use remark plugins here
    .process(sourceMarkdown);

  document.querySelector("#result").textContent = String(file);
  document.querySelector("#error").textContent = JSON.stringify(
    file.messages,
    null,
    4
  );
}

main().catch((error) => {
  document.querySelector("#error").textContent = error;
});
```

</details>

Without more detail I'm can only offer some general suggestions:

1. make sure you are using the latest versions of `remark` and `remark-lint-no-unused-definitions` this may be a bug that was already fixed, or caused by mixing incompatible versions.
2. if the line numbering shown above is part of the content, remove it, markdown definitions cannot have leading content on the same line
3. it could be related to adminitions but in a way not shown in your example. If so you may need an admonition parser, there is one for `remark` https://github.com/elviswolcott/remark-admonitions but it is behind on versions and is not yet compatible with latest (see https://github.com/elviswolcott/remark-admonitions/issues/27 and https://github.com/elviswolcott/remark-admonitions/issues/49, the repo could use some help if you want to look into supporting latest remark)

**Comment by samjcombs** [comment]
thanks for the reply. In response to your questions:

1. Yes Im using the latest versions
2. Line numbering was added just for that example to illustrate where the error is being reported
3. Yes likely a gap when using admonitions.. for now ill just ignore these but yeah good idea to look into supporting that library. 

ill try to recreate this error state in a codesandbox for reference.

**Comment by ChristianMurphy** [comment]
Thanks for the confirmation.
That isn't something which can't necessarily be fixed here.

`remark-lint` works off of what is parsed.

`remark` itself parses commonmark.
And plugins layer features on top of that, like the `remark-gfm` or `remark-admonitions` parser extensions.

If the markdown flavor you tell remark to parse is different from the flavor the content actually is, the linter won't be able to make sense of the content.

Happy to answer further questions in the discussion forum https://github.com/orgs/remarkjs/discussions
And if you can replicate the issue in plain and valid commonmark, feel free to add details.

In the meantime closing this out as something which requires admonition fixes rather than linter changes.
